docs/docsite/rst/dev_guide/developing_inventory.rst

```diff
@@ -1,21 +1,220 @@
 .. _developing_inventory:
 
-Developing Dynamic Inventory Sources
-====================================
+Developing Dynamic Inventory
+============================
 
 .. contents:: Topics
    :local:
 
-As described in :ref:`dynamic_inventory`, Ansible can pull inventory information from dynamic sources, including cloud sources. You can also create a new dynamic inventory provider by creating a script or program that can output JSON in the correct format when invoked with the proper arguments. There is no restriction on the language used for creating a dynamic inventory provider.
+As described in :ref:`dynamic_inventory`, Ansible can pull inventory information from dynamic sources,
+including cloud sources, using the supplied :ref:`inventory plugins <Inventory_Plugins>`.
+If the source you want is not currently covered by existing plugins, you can create your own as with any other plugin type.
+
+In previous versions you had to create a a script or program that can output JSON in the correct format when invoked with the proper arguments.
+This is still supported via the :ref:`script inventory plugin <script_inventory>` and there is no restriction on the programming language used.
+The disadvantages are that they cannot use integrated functionality like the :ref:`inventory plugins <Inventory_Plugins>` can and then need to implement these features themselves. i.e caching, configuration management, dynamic variable and group composition, etc.
+
+
+.. _inventory_sources:
+
+Inventory sources
+-----------------
+
+Inventory sources are strings (i.e what you pass to ``-i`` in the command line).
```

Comment:
This has me a bit confused. Here we're staying inventory sources are strings, but below we're saying they can be strings or files or scripts. I think I know what you're saying here - here's how I would express it: 
"Inventory sources are lists of managed nodes (i.e. what you pass to ``-i`` at the command line). You can pass your inventory source as a comma-separated list, a file path, a URI, or anything your plugin can read as input. Select the correct plugin for your inventory source: the host list plugin reads strings, the yaml and ini plugins read data files, the virtualbox and constructed plugins read configuration files, and the script plugin reads scripts or executable files."

Hm, thinking about it, I might put a little table here with two columns, one for the plugin and one for the input(s) it reads.